apps/blockchain/test/blockchain/interface/account_interface_test.exs

```diff
@@ -2,4 +2,233 @@ defmodule Blockchain.Interface.AccountInterfaceTest do
+
+      {:ok, found_code} =
+        state
+        |> Account.reset_account(address)
+        |> Account.put_code(address, code)
+        |> AccountInterface.new()
+        |> AccountInterface.machine_code(address)
+
+      assert found_code == code
+    end
+  end
+
+  describe "clear_balance/2" do
+    test "clears account's balance", %{state: state} do
+      address = <<1>>
+
+      account = %Account{
+        nonce: 5,
+        balance: 10,
+        storage_root: <<0x00, 0x01>>,
+        code_hash: <<0x01, 0x02>>
+      }
+
+      {found_account, _} =
+        state
+        |> Account.put_account(address, account)
+        |> AccountInterface.new()
+        |> AccountInterface.clear_balance(address)
+        |> AccountInterface.account(address)
+
+      assert found_account.balance == 0
+    end
+  end
+
+  describe "reset_account/2" do
+    test "resets account", %{state: state} do
+      address = <<1>>
+
+      account = %Account{
+        nonce: 5,
+        balance: 10,
+        storage_root: <<0x00, 0x01>>,
+        code_hash: <<0x01, 0x02>>
+      }
+
+      {found_account, _} =
+        state
+        |> Account.put_account(address, account)
+        |> AccountInterface.new()
+        |> AccountInterface.reset_account(address)
+        |> AccountInterface.account(address)
+
+      assert found_account == %Account{}
+    end
+  end
+
+  describe "add_wei/2" do
+    test "adds wei to account's balance", %{state: state} do
+      address = <<1>>
+
+      account = %Account{
+        nonce: 5,
+        balance: 10,
+        storage_root: <<0x00, 0x01>>,
+        code_hash: <<0x01, 0x02>>
+      }
+
+      {found_account, _} =
+        state
+        |> Account.put_account(address, account)
+        |> AccountInterface.new()
+        |> AccountInterface.add_wei(address, 100)
+        |> AccountInterface.account(address)
+
+      assert found_account.balance == 110
+    end
+  end
+
+  describe "account/2" do
+    test "fetches account from cache", %{state: state} do
+      account = %Account{
+        nonce: 5,
+        balance: 10,
+        storage_root: <<0x00, 0x01>>,
+        code_hash: <<0x01, 0x02>>
+      }
+
+      address = <<1>>
+      code = <<5>>
+      cache = %Cache{accounts_cache: %{address => {account, code}}}
+
+      account_interface =
+        state
+        |> Account.reset_account(address)
+        |> AccountInterface.new(cache)
+
+      assert AccountInterface.account(account_interface, address) == {account, code}
+    end
+
+    test "fetches account from storage", %{state: state} do
+      address = <<1>>
+
+      account_interface =
+        state
+        |> Account.reset_account(address)
+        |> AccountInterface.new()
+
+      {result, nil} = AccountInterface.account(account_interface, address)
+
+      assert result == %Blockchain.Account{
+               balance: 0,
+               code_hash:
+                 <<197, 210, 70, 1, 134, 247, 35, 60, 146, 126, 125, 178, 220, 199, 3, 192, 229,
```

Comment:
might be nice to name this `empty_keccak` or something as a helper method or a module variable